src/test/java/edu/carroll/cs389application/service/LoginServiceImplTest.java

```diff
@@ -0,0 +1,55 @@
+package edu.carroll.cs389application.service;
+
+import edu.carroll.cs389application.web.form.LoginForm;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.springframework.test.util.AssertionErrors.assertNotNull;
+import static org.springframework.test.util.AssertionErrors.assertTrue;
+
+
+/**
+ * This class contains unit tests for the LoginServiceImpl class.
+ */
+@SpringBootTest
+public class LoginServiceImplTest {
+
+    private static final String username = "testuser";
+
+    @Autowired
+    private UserService loginService;
+
+    /**
+     * Ensures that the loginService is injected.
+     * Also creates a test user in the database.
+     */
+    @BeforeEach
+    public void doBeforeTest() {
+        assertNotNull("loginService must be injected", loginService);
+
+        //Make sure that "testuser" is in database before our validateExistingUserSuccessfulTest
```

Comment:
I don't believe this comment is valid, as I don't think the validateUsername method does anything to insert a user.  And, it's return code is not checked, so this entire method is almost a no-op that just slows down testing and I think can be removed.